pkg/client/client_test.go

```diff
@@ -192,16 +232,34 @@ var _ = Describe("Client", func() {
 			close(done)
 		})
 
-		It("should fail if the object does not pass server-side validation", func() {
-			// TODO(seans3): implement these
+		It("should fail if the object does not pass server-side validation", func(done Done) {
+			cl, err := client.New(cfg, client.Options{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cl).NotTo(BeNil())
+
+			By("creating the pod, since required field Containers is empty")
+			err = cl.Create(context.TODO(), pod)
+			Expect(err).To(HaveOccurred())
+
+			close(done)
 		})
 
 		It("should fail if the object cannot be mapped to a GVK", func() {
-			// TODO(seans3): implement these
+			By("creating client with empty Scheme")
+			emptyScheme := runtime.NewScheme()
+			cl, err := client.New(cfg, client.Options{Scheme: emptyScheme})
```

Comment:
We can make it more closer to the actual user's usecase where: we create the `cl` without specifying the scheme so that client is initialized with default scheme which understands the core-type resources and then use a custom struct (we can define one in this test file) for Create method and it should fail.

Reply:
Its good to check that the provided scheme is actually used though :)

Reply:
I'll add a TODO for this. I don't want this PR to take much longer, so I'll address further checks in a future PR.